lib/posthog/feature_flags.rb

```diff
@@ -42,96 +77,288 @@ def is_feature_enabled(key, distinct_id, default_result = false, groups = {})
         end
       end
 
-      return default_result if !feature_flag
+      if !feature_flag.nil?
+        begin
+          response = _compute_flag_locally(feature_flag, distinct_id, groups, person_properties, group_properties)
+          logger.debug "Successfully computed flag locally: #{key} -> #{response}"
+        rescue InconclusiveMatchError => e
+          logger.debug "Failed to compute flag #{key} locally: #{e}"
+        rescue StandardError => e
+          logger.error "Error computing flag locally: #{e}"
+        end
+      end
 
-      flag_rollout_pctg =
-        if feature_flag['rollout_percentage']
-          feature_flag['rollout_percentage']
-        else
-          100
+      if response.nil?
+        begin
+          flags = get_feature_variants(distinct_id, groups, person_properties, group_properties)
+          response = flags[key]
+          logger.debug "Successfully computed flag remotely: #{key} -> #{response}"
+        rescue StandardError => e
+          logger.error "Error computing flag remotely: #{e}"
+          response = default_result
         end
-      if feature_flag['is_simple_flag']
-        return is_simple_flag_enabled(key, distinct_id, flag_rollout_pctg)
+      end
+
+      return response
```

Comment:
```suggestion
      response
```

Using return usually signifies an early return in ruby